libvast/src/system/source.cpp

```diff
@@ -0,0 +1,196 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2021 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "vast/system/source.hpp"
+
+#include "vast/fwd.hpp"
+
+#include "vast/defaults.hpp"
+#include "vast/detail/assert.hpp"
+#include "vast/error.hpp"
+#include "vast/expression.hpp"
+#include "vast/format/reader.hpp"
+#include "vast/logger.hpp"
+#include "vast/schema.hpp"
+#include "vast/system/actors.hpp"
+#include "vast/system/instrumentation.hpp"
+#include "vast/system/source_common.hpp"
+#include "vast/system/status_verbosity.hpp"
+#include "vast/table_slice.hpp"
+
+#include <caf/downstream.hpp>
+#include <caf/event_based_actor.hpp>
+#include <caf/settings.hpp>
+#include <caf/stateful_actor.hpp>
+
+#include <chrono>
+
+namespace vast::system {
+
+caf::behavior
+source(caf::stateful_actor<source_state>* self, format::reader_ptr reader,
```

Comment:
You seem to have reverted a fair amount of recent changes to this function. Please take the most recent version of `source.hpp` for the conversion, and then do the same thing again for the datagram source.

Reply:
@tobim f0029f4 is still incomplete and there are still changes missing.